### PR TITLE
Feature/saving loading

### PR DESCRIPTION
**Pull request by lkrombeen**
You can save files to an xml and load thse files. Also possible to modify them since they are not binary. Includes the save and load state + functionality + transitions.

### CODE REVIEW

File: IRescue/Unity/Assets/Scripts/Unity/ObjectPlacing/States/LoadState.cs [hunk 1]
```diff
@@ -4,17 +4,208 @@
...
+        /// Initialized the load select frame
+        /// </summary>
+        /// <param name="entryHeight">the height of an entry</param>
+        /// <param name="frameWidth">The width of the frame of the load scroll panel</param>
+        /// <returns>The Object select frame game object</returns>
+        private GameObject InitLoadScrollPane(int entryHeight, int frameWidth)
+        {
+            GameObject scollButton = UnityEngine.Object.Instantiate(Resources.Load<GameObject>("Prefabs/Buttons/LoadScrollButton"));
+            string[] filePaths = Directory.GetFiles(Directory.GetCurrentDirectory() + @"\\Saves\\");
+            FileSystemInfo[] files = new DirectoryInfo(Directory.GetCurrentDirectory() + @"\\Saves\\").GetFileSystemInfos();
```
Comment (yappel) [inline IRescue/Unity/Assets/Scripts/Unity/ObjectPlacing/States/LoadState.cs:183]: It looks like this variable is never used.

File: IRescue/Unity/Assets/Scripts/Unity/ObjectPlacing/States/SaveState.cs [hunk 2]
```diff
@@ -4,17 +4,143 @@
...
+
     /// <summary>
     /// State when saving the game.
     /// </summary>
     public class SaveState : AbstractState
     {
         /// <summary>
+        /// The folder where the save files are located from Unity/
+        /// </summary>
+        private const string SaveFile = "Saves/";
```
Comment (yappel) [inline IRescue/Unity/Assets/Scripts/Unity/ObjectPlacing/States/SaveState.cs:21]: We have a const for the save files folder in 2 different files, is there a way we could just have one place where we define the save files folder?

**Comment by yappel** [comment]
capture
Every time you press the open file button it opens a new window, even when there is already a window present.
I also get a exception when trying to open the file listed in the open save file window. I think the save file folder is not created correctly on my machine.
capture

**Comment by yappel** [comment]
There is a bug where all UI elements get removed. I managed to get in this situation by: Starting app - press open file- click 3 times on the save file in the list - load the selected file - press the save button.

**Comment by yappel** [comment]
I cant test with the actual meta, but the application is extremely buggy when ran with the MouseLocalizer.

**Comment by lkrombeen** [comment]
I have fixed the bugs. One issue remains with the box of the load state: If you got < 3 saves, the buttons will nog begin from the top of the panel, but from the center (since the content grows from the center until it overflows)

image

The panel works fine with more files despite the funky look above. I will try to fix it tomorrow

**Comment by lkrombeen** [comment]
Fixed everything, ready for a 2nd check

**Comment by yappel** [comment]
When I load a file and try to edit an object that is not the default object, nothing happens. The buttons dont perform actions after being pressed. Only the "back" and "delete" button behave like they should.

When I alter a loaded file (add an extra building for example) and press the save button (even a few times) the changes are not saved to the (or any) file.

**Comment by lkrombeen** [comment]
Can you elaborate on the buttons?

**Comment by lkrombeen** [comment]
Fixed everything and even more

**Pull request merged**